tests/unit/s2n_handshake_test.c

```diff
@@ -434,6 +434,30 @@ int main(int argc, char **argv)
         }
     }
 
+    /* Ensure that a handshake can be performed after all non-standard file descriptors are closed */
+    {
+        long max_file_descriptors = sysconf(_SC_OPEN_MAX);
+        EXPECT_SUCCESS(max_file_descriptors);
+        EXPECT_TRUE(max_file_descriptors < INT_MAX);
+        for (int fd = STDERR_FILENO + 1; fd < max_file_descriptors; fd++) {
+            close(fd);
+        }
```

Comment:
You could put this test in a fork and close all file descriptors. Might be simpler / more reliable.

And do you actually need max_file_descriptors here, since you're not actually checking the return value of close?

Reply:
I moved this test to inside of a fork so all of the file descriptors can be closed. I tried looping up to INT_MAX, but this took a very long time, so I kept the sysconf call to retrieve the maximum number of open files.